app/(app)/vehicles/index.tsx

```diff
@@ -110,32 +123,27 @@ const VehiclesScreen = () => {
         ),
       }}
     >
-      <ScreenContent>
-        {defaultVehicle ? (
-          <Field label={t('VehiclesScreen.myDefaultVehicle')}>
-            <VehicleRow
-              vehicle={defaultVehicle}
-              onContextMenuPress={() => handleContextMenuPress(defaultVehicle.id)}
-            />
-          </Field>
-        ) : null}
-
-        {vehicles.length > (defaultVehicle ? 1 : 0) ? (
-          <Field className="flex-1" label={t('VehiclesScreen.myOtherVehicles')}>
-            <FlatList
-              data={vehicles.filter(({ isDefault }) => !isDefault)}
-              keyExtractor={({ id }) => id.toString()}
-              ItemSeparatorComponent={() => <Divider dividerClassname="bg-transparent h-1" />}
-              renderItem={({ item }) => (
-                <VehicleRow
-                  vehicle={item}
-                  onContextMenuPress={() => handleContextMenuPress(item.id)}
-                />
-              )}
-            />
-          </Field>
-        ) : null}
-      </ScreenContent>
+      <View className="flex-1">
+        <SectionList
+          sections={sections}
+          stickySectionHeadersEnabled={false}
+          renderSectionHeader={({ section }) =>
+            section.data.length > 0 ? (
+              // Add padding only if it's not the first section, TODO find cleaner solution
+              <View className={cn({ 'mt-5': section.title !== sections[0].title })}>
+                <Typography variant="default-bold">{section.title}</Typography>
+              </View>
+            ) : null
+          }
+          className="p-5"
+          renderItem={({ item }) => (
+            <VehicleRow vehicle={item} onContextMenuPress={() => handleContextMenuPress(item.id)} />
+          )}
+          ItemSeparatorComponent={() => <Divider dividerClassname="bg-transparent h-1" />}
+          // SectionSeparatorComponent is added above and below section header, so we add only h-1 height and use workaround with top margin in renderSectionHeader
```

Comment:
by using this approach I assume we cannot use it directly on section header, right?